tests/west/manifest/test_manifest.py

```diff
@@ -9,11 +9,27 @@
 import pytest
 import yaml
 
+from west import config
 from west.manifest import Manifest, Defaults, Remote, Project, \
-    MalformedManifest
+    SpecialProject, MalformedManifest
 
 THIS_DIRECTORY = os.path.dirname(__file__)
 
+@pytest.fixture
+def config_file_project_setup(tmpdir):
```

Comment:
This fixture doesn't provide the correct configuration file in all cases: in `test_self_tag`, the west.config.config object is missing a `path=mainproject`. You might want to parametrize the fixture or just use `patch` to substitute a test-specific mock for the `west.config.config` value.

Reply:
That doesn't matter in this test, as the `path` is taken from the manifest when creating a list of projects, and not the conf file.

The `path` in the conf file is used to locate the manifest file, although this is not used in the test, as the content of the manifest is fed directly `Manifest.from_data(yaml.safe_load(content))`

Reason for taking path form manifest file when creating the list of projects is that in future it will allow us to detect if `path` inside manifest changes, and thus `config_file.path != manifest_file.self_path`, as example after a `git reset` or local modifications.